lib/ui/dropdown/dropdown.js

```diff
@@ -172,16 +180,41 @@
 
     this.getMultiSelected = function(viewValue) {
 
-      var options = this.collection($scope);
       var indices = [];
 
-      _.each(viewValue, function(savedObject) {
-        var index = _.findIndex(options, function(value) {
-          var temp = angular.copy(savedObject); // remove hashkeys for comparison
-          return _.matches(temp)(value);
+      if($element.get(0).tagName.toLowerCase() !== 'input') {
+        var options = this.collection($scope);
+
+        _.each(viewValue, function(savedObject) {
+          var index = _.findIndex(options, function(value) {
+            var temp = angular.copy(savedObject); // remove hashkeys for comparison
+            return _.matches(temp)(value);
+          });
+          indices.push(index);
         });
-        indices.push(index);
-      });
+
+      } else {
+
+        var inputViewValues = this.ngModel.$modelValue;
+
+        _.each(inputViewValues, function(savedObject) {
+
+          if(_.isUndefined(savedObject.id) ) {
+
+            if(!_.isEmpty(savedObject[self.options.idSubstitute])) {
+
+              savedObject.id = savedObject[self.options.idSubstitute];
+
+            } else {
+              throw new Error('dropdown list must have a id or a alternative value to use as a id');
```

Comment:
I would probably throw the error much earlier during the init phase of the component.